builtin/providers/aws/resource_aws_vpc.go

```diff
@@ -102,6 +106,18 @@ func resourceAwsVpcCreate(d *schema.ResourceData, meta interface{}) error {
 			d.Id(), err)
 	}
 
+	// Link DHCP Options Set if specified
+	if _, ok := d.Get("dhcp_options_id").(string); ok {
```

Comment:
So we also need to do update the association to the "default" if the dhcp_options_id is removed later. Plus, we must compute this value for cases where it is not provided.